Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/GridObjectCollection.cs

```diff
@@ -0,0 +1,482 @@
+        }
+
+        public float Width
+        {
+            get { return width; }
+        }
+
+        public float Height
+        {
+            get { return height; }
+        }
+        #endregion
+
+
+        /// <summary>
+        /// Overriding base function function for laying out all the children when UpdateCollection is called.
+        /// </summary>
+        protected override void LayoutChildren()
+        {
+
+            columns = Mathf.CeilToInt((float)NodeList.Count / Rows);
+            width = columns * CellWidth;
+            height = Rows * CellHeight;
+            halfcell = new Vector2(CellWidth / 2f, CellHeight / 2f);
+            circumferenc = 2f * Mathf.PI * Radius;
+
+
+            int cellCounter = 0;
+            float startOffsetX;
+            float startOffsetY;
+
+            Vector3[] nodeGrid = new Vector3[NodeList.Count];
+            Vector3 newPos = Vector3.zero;
+            Vector3 newRot = Vector3.zero;
+
+            // Now lets lay out the grid
+            startOffsetX = (columns * 0.5f) * CellWidth;
+            startOffsetY = (Rows * 0.5f) * CellHeight;
+
+            cellCounter = 0;
+
+            // First start with a grid then project onto surface
+            switch (layout)
+            {
+                case LayoutType.ColumnThenRow:
+                default:
+                    for (int c = 0; c < columns; c++)
+                    {
+                        for (int r = 0; r < Rows; r++)
+                        {
+                            if (cellCounter < NodeList.Count)
+                            {
+                                nodeGrid[cellCounter] = new Vector3((c * CellWidth) - startOffsetX + halfcell.x, -(r * CellHeight) + startOffsetY - halfcell.y, 0f) + (Vector3)((NodeList[cellCounter])).Offset;
+                            }
+                            cellCounter++;
+                        }
+                    }
+                    break;
+
+                case LayoutType.RowThenColumn:
+                    for (int r = 0; r < Rows; r++)
+                    {
+                        for (int c = 0; c < columns; c++)
+                        {
+                            if (cellCounter < NodeList.Count)
+                            {
+                                nodeGrid[cellCounter] = new Vector3((c * CellWidth) - startOffsetX + halfcell.x, -(r * CellHeight) + startOffsetY - halfcell.y, 0f) + (Vector3)((NodeList[cellCounter])).Offset;
+                            }
+                            cellCounter++;
+                        }
+                    }
+                    break;
+
+            }
+
+            switch (SurfaceType)
+            {
+                case ObjectOrientationSurfaceType.Plane:
+                    for (int i = 0; i < NodeList.Count; i++)
+                    {
+                        newPos = nodeGrid[i];
+                        NodeList[i].transform.localPosition = newPos;
+                        switch (OrientType)
```

Comment:
With all these nested switch statements, let's call a method instead and pass any values it needs to it.